processor/src/decorators/mod.rs

```diff
@@ -148,6 +151,39 @@ impl Process {
 
         Ok(())
     }
+
+    /// Given a quadratic extension field element ( say a ) on stack top, this routine computes
+    /// multiplicative inverse of that element ( say b ) s.t.
+    ///
+    /// a * b = 1 ( mod P ) | b = a ^ -1, P = irreducible polynomial of degree 1 over F_q, q = 2^64 - 2^32 + 1
+    ///
+    /// Input on stack expected in following order
+    ///
+    /// [coeff_1, coeff_0, ...]
+    ///
+    /// While computed multiplicative inverse is put on advice provider in following order
+    ///
+    /// [coeff'_0, coeff'_1, ...]
+    ///
+    /// Meaning when a Miden program is going to read it from advice tape, it'll see
+    /// coefficient_0 first and then coefficient_1.
+    ///
+    /// Note, in case input operand is zero, multiplicative inverse is returned to be zero.
+    fn inject_ext2_inv_result(&mut self) -> Result<(), ExecutionError> {
+        let coef0 = self.stack.get(1);
+        let coef1 = self.stack.get(0);
+
+        let elm = Ext2Element::new(coef0, coef1);
+        let inv_elm = elm.inv();
```

Comment:
We should check if `elem` is `ZERO` before trying to invert it, and if it is return a "divide by zero" error.